src/app/auth.context.tsx

```diff
@@ -0,0 +1,97 @@
+'use client';
+
+import React, { createContext, useEffect, useState } from 'react';
+import { IMonth, IUser } from '@/src/types';
+import { getUserMonths, verify } from '@/src/app/api';
+import { useRouter } from 'next/navigation';
+
+export interface IAppContext {
+  user: IUser | null;
+  isLoggedIn: boolean;
+  isLoadingContext: boolean;
+  userMonths: IMonth[];
+  storeToken(token: string): void;
+  authenticateUser(): void;
+  logOutUser(): void;
+}
+
+const AuthContext = createContext<IAppContext | null>(null);
+
+function AuthProviderWrapper({ children, allMonths }: { children: React.ReactNode; allMonths: IMonth[] }) {
+  const [isLoggedIn, setIsLoggedIn] = useState<boolean>(false);
+  const [isLoadingContext, setIsLoadingContext] = useState<boolean>(true);
+  const [userMonths, setUserMonths] = useState<IMonth[]>([]);
+  const [user, setUser] = useState<IAppContext['user'] | null>(null);
+  const router = useRouter();
+  const storeToken = (token: string) => {
+    localStorage.setItem('authToken', token);
+  };
```

Comment:
You don't need to create this function. This is optional but i would use `localStorage.setItem(TOKEN_FROM_ENUM, token);` just to follow the pattern